### ISSUE TITLE
Connection Opportunities "Activity Added" Workflow Trigger Not Saving Activity Type

### ISSUE THREAD

**Issue by NuclearTacos** [SparkDevNetwork/Rock#2029]
### Description
When editing a Connection Opportunity, and you set a workflow to be triggered by the 'Activity Added', the Activity Type box doesn't save your selection. 

You can hit save and then edit the same trigger and the box is empty, you can save, save the Connection Opportunity, edit it again, edit trigger and it'll be blank.  Also, workflows don't seem to fire off the trigger in testing on my Install

### Steps to Reproduce
1. Home page -> Connections -> gear in top right -> Connection Type -> Any Connection Opportunity
1. scroll down -> Workflows -> Plus -> "Launch Workflow When" = Activity Added - "Workflow Type" = any Workflow Type - "Activity Type" = anything, even blank since it's not required.
1. add an activity to a Connection Request of that Connection Opportunity

**Expected behavior:**

I expected selecting an Activity Type would allow me to trigger a Workflow when an Activity of that Activity Type was saved to the Connection, and I expected that an empty Activity Type would trigger for ANY saved Activity

**Actual behavior:**

No workflows are fired by that kind of trigger.

NOTE: It may be worth mentioning that these sorts of triggered events will be put into the Transaction Queue (as seen in the `Rock Information` Diagnostics tab; see screenshot below) and are generally executed within a minute or two.

![image](https://user-images.githubusercontent.com/383807/72921636-e50ba600-3d08-11ea-9e82-3fc39b8e77c1.png)

### Versions
* **Rock Version:** 6.1 (mine), 6.2 (demo)
* **Client Culture Setting** [en-us]

Diagnostic data attached for good measure.
[RockDiagnostics-ROCKPRODUCTION.txt ](https://github.com/SparkDevNetwork/Rock/files/780945/RockDiagnostics-ROCKPRODUCTION.txt)

**Comment by SteveSwaringen** [comment]
I'm having a similar problem. Trying to trigger a workflow when a Connection Request Connection Status is changed. The "From" and "To" statuses don't stick in the UI and don't appear to be effected (meaning the workflow doesn't trigger when the status changes).

**Comment by rutledgek** [comment]
This has been an open low priority bug since 2017.  I am using 9.4.  Will it be fixed or will the feature be removed as it makes activate on activity added nonfunctional.

**Comment by nairdo** [comment]
Thanks for the feedback everyone! We're pulling this one up from the depths and putting it into the next up queue.

**Comment by ethan-sparkdevnetwork** [comment]
Tested on 1.9.5.0 Using these steps I was able successfully run a workflow based on triggers in a Connection Request.

1. Created a simple workflow "Test Workflow" that sends an email from Rock to me.
2. Edited the connection type (Admin Tool --> Connections. Click the gear on the far right, click Involvement, click Edit.
3. Scroll to the bottom of the connection type block and expand Workflows
4. Click the plus sign to add a workflow
5. Assign "Test Workflow" as the Workflow Type
6. Set Launch Workflow When to Request STarted
7. Save and Save
8. Login as Cindy
9. Connect for the Ushers opportunity (email from workflow is sent)
10. As admin make changes to the workflow conditions and then a matching change to the opportunity request (email from workflow is sent)

Also the disappearing selection info in the From and To boxes was not observed.

**Comment by ethan-sparkdevnetwork** [comment]
@rutledgek Can you give me more info on the specific issue you are having? If you follow the steps I did does the workflow run?

**Comment by NuclearTacos** [comment]
@ethan-sparkdevnetwork The specific issue we're seeing was not with the "Request Started" But with "Activity added" in your step 6

**Comment by NuclearTacos** [comment]
FWIW, I just tested it again and it worked for me.  @rutledgek, would you be able to verify this isn't a problem for you anymore?

**Comment by nairdo** [comment]
@NuclearTacos @SteveSwaringen @rutledgek I have also not been able to reproduce either of the bugs you have explained above using Rock 9.5/9.4. _(Steve, when I try setting the From and To drop-downs they are working/saving fine.)_  I've also tried to locate where the fix may have happened, but I have not been able to find anything. 

I updated the original problem description above, but I'll mention it here too:  These sorts of triggered events will be put into the Transaction Queue (as seen in the Rock Information Diagnostics tab; see screenshot below) and are generally executed within a minute or two. In the example Steve illustrates, the item is called `ConnectionRequestChangeTransaction`: 

![image](https://user-images.githubusercontent.com/383807/72923053-7ed45280-3d0b-11ea-9697-06f13be4e077.png)


(When you're troubleshooting this issue, make sure the queue is empty before you try to find the triggered workflow.)
